embulk-input-s3/src/main/java/org/embulk/input/s3/HttpProxy.java

```diff
@@ -0,0 +1,133 @@
+package org.embulk.input.s3;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Optional;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URLDecoder;
+import java.util.Map;
+
+/**
+ * HttpProxy is config unit for Input/Output plugins' configs.
+ *
+ * TODO
+ * This unit will be moved to embulk/embulk-plugin-units.git.
+ */
+public class HttpProxy
+{
+    private final String host;
+    private final Optional<Integer> port;
+    private final boolean useSsl;
+    private final Optional<String> user;
+    private final Optional<String> password; // TODO use SecretString
+
+    @JsonCreator
+    public HttpProxy(
+            @JsonProperty("host") String host,
+            @JsonProperty("port") Optional<Integer> port,
+            @JsonProperty("use_ssl") boolean useSsl,
+            @JsonProperty("user") Optional<String> user,
+            @JsonProperty("password") Optional<String> password)
+    {
+        this.host = host;
+        this.port = port;
+        this.useSsl = useSsl;
+        this.user = user;
+        this.password = password;
+    }
+
+    public String getHost()
+    {
+        return host;
+    }
+
+    public Optional<Integer> getPort()
+    {
+        return port;
+    }
+
+    public boolean useSsl()
+    {
+        return useSsl;
+    }
+
+    public Optional<String> getUser()
+    {
+        return user;
+    }
+
+    public Optional<String> getPassword()
+    {
+        return password;
+    }
+
+    /**
```

Comment:
While we don't have env-based configuration, we may not want to keep the code. How about adding them later when we add env-based configurations?